### ISSUE TITLE
docker-compose push doesn't work while same image is successfully pushed using docker

### ISSUE THREAD

**Issue by amitsehgal** [docker/compose#3929]
```
Mac:

docker-compose -f docker-compose-build.yml push
  Pushing xyz (abc.com/def/xyz:latest)...
  The push refers to a repository [abc.com/def/xyz]
  blah
  blah
  ERROR: unauthorized: authentication required
  
  docker push abc.com/def/xyz
  The push refers to a repository [abc.com/def/xyz]
  blah
  blah
  latest: digest: sha256:xxxxxxxxx size: xxxxxxx
  
  compose version:
  docker-compose version 1.8.0, build f3628c7
  docker-py version: 1.9.0
  CPython version: 2.7.9
  OpenSSL version: OpenSSL 1.0.2h  3 May 2016
  
  
  
  Client:
 Version:      1.12.0
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   8eab29e
 Built:        Thu Jul 28 21:15:28 2016
 OS/Arch:      darwin/amd64

Server:
 Version:      1.12.0
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   8eab29e
 Built:        Thu Jul 28 21:15:28 2016
 OS/Arch:      linux/amd64 
```

**Comment by shin-** [comment]
Are you using the `osxkeychain` credentials store?

**Comment by shin-** [comment]
Should be fixed in 1.8.1
